### DISCUSSION TITLE
Installing this

### DISCUSSION THREAD

**Discussion by Tlerpiniere** [tujson/GlassEcho#5] [Q&A]
Amazing work on getting hardware back up and running. I'm wondering if it were possible to give a quick overview how to install this to the phone / glass for those of us with limited development knowledge. The reddit glass group will be desperate for this project to work!

**Comment by tujson** [comment]
Sure! Once GlassEcho reaches a more stable and user-friendly state, I'll be publishing APKs through GitHub's releases functionality. This way people can directly download the APK.

But meanwhile I'd be happy to include a "INSTALLING.md" file with detailed instructions on how to install the app. I'll get that up sometime this weekend.

**Reply by crashdot** [reply]
Hi, I wonder if there is a configuration requirement in Android Studio that may not be present by default. My devices are attached to Android Studio, as I can see their activity in Logcat and browse them in Device Explorer, but I can find no way to run the project on either device, as they do not appear in the device drop down. Also, the configuration drop down says "Add Configuration" so the run arrow is greyed out, but I am unable to determine what to do in order to create a working configuration. If I attempt to create one, it says "AndroidManifest.xml doesn't exist or has incorrect root tag". When I look up how to resolve the AndroidManifest error, I am directed to click a "sync project to gradle" option, but there is no such option or icon to be found. Any help would be greatly appreciated.

**Comment by Tlerpiniere** [comment]
Honestly thay would be amazing. Thank you so so much! I'll keep a look at
on my emails :)

**Comment by Tlerpiniere** [comment]
Just an update from my end, after mammoth amounts of research on how to
sideload, and issues with adb showing the device as offline, I've managed
to install on my glass!

Unfortunatly no notifications want to pass through to it from my Fold 2,
I'll give it another test on an older device later to see if it's the phone
that's the issue.

Thanks so much again for all your hard work,
Tom

**Reply by tujson** [reply]
Interesting, I don't have any Samsung devices with me. Do you know what version of Android it's running?
Also, just to double check, you have the notification slider on in the phone app's "notifications" tab?

**Comment by Tlerpiniere** [comment]
Hi Jason,

Attached are my android details, android 11 my end. The notification tab
has everything selected. I tried with several apps.

I have a few old phones at home so I'll give that a bash :)

Thanks!

**Comment by parkernbrown** [comment]
First things first, great work on this! I'm excited to see where it goes!

I was able to get the app loaded onto my phone, but when I try to load it onto my Glass, Android Studio claims that it can't find ADB. I've tried uninstalling and reinstalling the platform tools, adding a system variable, killing ADB and starting it again..... Nothing is working. Do you have any advice or a workaround?

Thanks!

**Reply by parkernbrown** [reply]
NEVERMIND!

I figured it out!

https://stackoverflow.com/questions/20435778/google-glass-is-not-listed-as-android-device-by-adb#:~:text=Turns%20out%20you%20can't,(or%20something%20like%20that).
